bali/application.py

```diff
@@ -77,16 +81,25 @@ async def _launch_rpc(self):
 
     def _launch_event(self):
         from .events import handle
+        event_handler = self.kwargs.get('event_handler')
+        if not event_handler:
+            raise Exception('event_handler not provided')
         while True:
-            handle()
+            try:
+                handle()
```

Comment:
I'm afraid that it will cause an endless loop if there is a exception in `handle()`

Reply:
when an exception occurred，other event should not be effect

Reply:
What I mean is the program will output errors like crazy if there is exception raised in `handle`, not event handler functions.

What about the exception raised in the earlier block. Just like it can't connect to the AMQP server. What will happen in this case, maybe it will continuous output of the error log.

```python
def handle():
    groups = groupby(REGISTER_EVENT_CALLBACKS, key=lambda x: str(x))
    for k, items in groups:
        items = list(items)
        with get_connection(amqp_address=items[0].connection) as conn:
            with conn.Consumer(
                queues=[items[0].queue],
                accept=['json'],
                callbacks=[i.callback for i in items],
                auto_declare=True
            ) as consumer:
                try:
                    conn.drain_events(timeout=2)
                except socket.timeout:
                    pass
                except Exception as e:
                    raise e
    return True
```